File: typescript/src/model/current_geo_location.ts
import { Container } from "./container";

export class CurrentGeoLocation {
    constructor(
        public container: Container | null = null
    ) { }

    toMap(): Record<string, any> {
        return {
            'container': this.container?.toMap() ?? null,
        };
    }

    static fromMap(data: Record<string, any>): CurrentGeoLocation {
        if(data == null) {
            return new CurrentGeoLocation();
        }
        
        const containerData = data['container'];
        const container = containerData ? Container.fromMap(containerData) : null;
        return new CurrentGeoLocation(container);
    }
}

File: dart/lib/src/model/current_geo_location.dart
import 'package:openral_core/src/model/container.dart';
import 'package:openral_core/src/model/geo_coordinates.dart';
import 'package:openral_core/src/model/postal_address.dart';

class CurrentGeoLocation {
  Container? container;

  String? threeWordCode;

  PostalAddress? postalAddress;

  GeoCoordinates? geoCoordinates;

  String? plusCode;

  CurrentGeoLocation({
    this.container,
    this.postalAddress,
    this.threeWordCode,
    this.geoCoordinates,
    this.plusCode,
  });

  Map<String, dynamic> toMap() {
    return {
      'container': container?.toMap(),
      'postalAddress': postalAddress?.toMap(),
      '3WordCode': threeWordCode,
      'geoCoordinates': geoCoordinates?.toMap(),
      'plusCode': plusCode,
    };
  }
}
